### PR TITLE
Add fullScreen mode to Modal

### PR DESCRIPTION
**Pull request by Klynger**
#### What problem is this solving?

Mobile visualization wasn't good, now is a bit better.

#### How should this be manually tested?

[Workspace](https://modallayout--storecomponents.myvtex.com/) click in the buy button of a product and change the view from desktop to mobile in the devtools

#### Checklist/Reminders

- [X] Updated `README.md`.
- [X] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -47,9 +47,12 @@ WIP
...
 | `title` | `string`| The title of the modal | `undefined` |
-| `showCloseButton`| `boolean`| If it should show the close button | `true` |
-| `backdrop`| `BackdropMode` |  | `'clickable'` |
-| `titleTag` | TitleTag | Which tag the title element should render | `'h3'` |
+| `showCloseButton`| `ResponsiveValue<boolean> | boolean`| If it should show the close button | `true` |
+| `backdrop`| `ResponsiveValue<BackdropMode> | BackdropMode` | How the backdrop should be rendered | `'clickable'` |
+| `titleTag` | `TitleTag` | Which tag the title element should render | `'h3'` |
+| `fullScreen` | `ResponsiveValue<boolean> | boolean` | If the modal should be in full screen | `false` |
+
+You can know more about `ResponsiveValue` in the documentation of [responsive-values](https://github.com/vtex-apps/responsive-values).
```
Comment (kevinch) [inline docs/README.md]:
```suggestion
You can learn more about `ResponsiveValue` in the documentation of [responsive-values](https://github.com/vtex-apps/responsive-values).
```

File: react/Modal.tsx [hunk 2]
```diff
@@ -1,24 +1,23 @@
 import React from 'react'
 import { pick } from 'ramda'
+import styles from './styles.css'
```
Comment (kevinch) [inline react/Modal.tsx]: Nitpick: keep absolute and relative imports separated.

File: react/Modal.tsx [hunk 3]
```diff
@@ -64,7 +64,14 @@ const Modal: React.FC<Props> = props => {
     }
   }
 
-  const containerClasses = classnames(styles.containerCenter, handles.container, 'bg-base relative flex flex-column')
+  const containerClasses = classnames(
+    styles.containerCenter,
+    handles.container,
+    'bg-base relative flex flex-column',
+    {
+      [`${styles.fullScreenModal} w-100 mw-100 h-100 br0`]: fullScreen,
```
Comment (kevinch) [inline react/Modal.tsx]: Sounds a bit complicated imo.
```suggestion
      [styles.fullScreenModal]: fullScreen,
```

File: react/Modal.tsx [hunk 4]
```diff
@@ -64,7 +64,14 @@ const Modal: React.FC<Props> = props => {
     }
   }
 
-  const containerClasses = classnames(styles.containerCenter, handles.container, 'bg-base relative flex flex-column')
+  const containerClasses = classnames(
+    styles.containerCenter,
+    handles.container,
+    'bg-base relative flex flex-column',
```
Comment (kevinch) [inline react/Modal.tsx]:
```suggestion
    'bg-base relative flex flex-column w-100 mw-100 h-100 br0',
```

File: react/ModalTrigger.tsx [hunk 5]
```diff
@@ -17,13 +17,13 @@ const ModalTrigger: React.FC<{}> = ({ children }) => {
   }
 
   return (
-    <button
+    <div
       role="button"
       onClick={handleModalOpen}
       className={`${handles.triggerContainer} bg-transparent pa0 ma0 outline-0 bw0`}
```
Comment (kevinch) [inline react/ModalTrigger.tsx]: A div has no margin by definition. Does keeping `ma0` make sense?

File: docs/README.md [hunk 6]
```diff
@@ -47,9 +47,12 @@ WIP
...
 | `title` | `string`| The title of the modal | `undefined` |
-| `showCloseButton`| `boolean`| If it should show the close button | `true` |
-| `backdrop`| `BackdropMode` |  | `'clickable'` |
-| `titleTag` | TitleTag | Which tag the title element should render | `'h3'` |
+| `showCloseButton`| `ResponsiveValue<boolean> | boolean`| If it should show the close button | `true` |
+| `backdrop`| `ResponsiveValue<BackdropMode> | BackdropMode` | How the backdrop should be rendered | `'clickable'` |
+| `titleTag` | `TitleTag` | Which tag the title element should render | `'h3'` |
+| `fullScreen` | `ResponsiveValue<boolean> | boolean` | If the modal should be in full screen | `false` |
+
+You can learn more about `ResponsiveValue` in the documentation of [responsive-values](https://github.com/vtex-apps/responsive-values).
```
Comment (victorhmp) [inline docs/README.md]:
```suggestion
You can learn more about `ResponsiveValue` in the documentation of [responsive-values](https://vtex.io/docs/app/vtex.responsive-values).
```

File: react/Modal.tsx (hunk 4 above)
Comment (Klynger) [reply react/Modal.tsx]: These classes are part of the `fullScreen` visualization, so I can't apply them here

**Pull request merged**